splinepy/helpme/create.py

```diff
@@ -277,6 +284,134 @@ def from_bounds(parametric_bounds, physical_bounds):
     return bspline_box
 
 
+def determinant_spline(self, dmin=False, show=False):
+    """
+    Creates the jacobian determinant spline of the given spline object.
+    Only works if (parameter dimension == physical dimension) and non-rational splines
+    or rational splines with equal weights
+    (otherwise give B-Spline approximation of spline)!
+
+    references:
+    Mantzaflaris, A., Jüttler, B., Khoromskij, B. N., & Langer, U. (2017)
+    Limkilde, A., Evgrafov, A., Gravesen, J., & Mantzaflaris, A. (2021)
+
+    Parameters
+    ----------
+    dmin: bool, optional
+      returns dmin of determinant spline
+      dmin > 0 --> spline is not tangled
+      dmin <= 0 --> spline is COULD BE tangled
+    show: bool, optional
+      Shows determinant spline mapped onto original spline.
+
+    Returns
+    -------
+    detSp: BSpline
+      Spline which represents the jacobian determinant of the given spline object.
+    """
+
+    from splinepy import BSpline
+
+    # checks if dimensions are equal.
+    if self.dim != self.para_dim:
+        raise ValueError(
+            f"Dimension of parameter space and physical space ({self.whatami})"
+            "are not the same!"
+        )
+
+    # checks if spline is rational and if weights differ
+    # otherwise NURBS can be treated like regular BSpline
+
+    if self.is_rational and any(
+        np.around(np.array(self.weights), 10)
+        != np.round(np.array(self.weights)[0], 10)
+    ):
+        raise TypeError(
+            f"Spline is rational ({self.whatami}) with unequal weights!"
+            "Please give a valid Approximation (e.g. BSpline) of the spline."
+        )
+
+    # calculate degrees of det spline
+    deg_det = self.degrees * self.dim - 1
+    d = self.para_dim
+    # initialize lists for knot vectors
+    kvs_det = [[]] * d
+    unique_knots = [[]] * d
```

Comment:
```suggestion
    kvs_det = [None] * d
```
I'm guess this is not intended. this list will have reference to the same object d times. I see that you overwrite each element anyways. You also overwrite `unique_knots` completely d times.